src/lib/libeventloop.js

```diff
@@ -21,6 +21,13 @@ LibraryJSEventLoop = {
     }, timeout);
   },
 
+  $safeClearTimeout: (id) => {
+    // TODO: Track outstanding timeout IDs so that we can avoid calling
+    // runtimeKeepalivePop trice for the same timeout.
+    {{{ runtimeKeepalivePop() }}}
```

Comment:
Without this tracking, can't we consider the program to have exited at an earlier time than it did? That sounds more dangerous than the state before this PR IIANM?

Reply:
Yes, but only programs that call `emscripten_clear_timeout` with the ID of a timeout that has already fired.

Without this change it is not possible for programs that clear timeouts to exit.

With this change programs that clear invalid timeout ID might exit earlier than they expect.

Reply:
Perhaps I'll open a bug and make this change NFC.

Reply:
That sgtm. Early exits can be pretty dangerous, worse than not being able to exit, I'd say.

Reply:
Reverting this change and making this a test-only NFC